### PR TITLE
Segment identify refactor

### LINKED ISSUE #820: Add CC conversion events to segment
## Issue - Enhancement

Let's track more of the user conversion funnel.

1. When a user clicks on a trial conversion modal
1. When a user clicks to add their CC info
1. Whan a user successfully adds their CC info

Any other events I should consider?

- **Estimated effort:** 12 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

### PR DESCRIPTION
**Pull request by jordanh**
Closes #820

### CODE REVIEW

**Comment by jordanh** [comment]
@mattkrick @ackernaut I think this is GTG. Have a peek?

File: src/universal/modules/userDashboard/containers/CreditCardModal/CreditCardModalContainer.js [hunk 1]
```diff
@@ -88,7 +93,13 @@ class CreditCardModalContainer extends Component {
         stripeToken
       };
       const {error: cashayError} = await cashay.mutate('addBilling', {variables});
-      if (cashayError) throw new SubmissionError(cashayError);
+      if (cashayError) {
+        dispatch(
+          segmentEventTrack('addStripeBilling Cashay Error', {error: cashayError._error})
```
Comment (mattkrick) [inline src/universal/modules/userDashboard/containers/CreditCardModal/CreditCardModalContainer.js]: i'd prefer to throw this one & the success one from the server for 2 reasons:
- less logic on the client
- guaranteed to call

Comment (jordanh) [inline src/universal/modules/userDashboard/containers/CreditCardModal/CreditCardModalContainer.js]: I'll toss the error one, but let's keep the success one. It's key for the funnel metrics.

**Comment by jordanh** [comment]
What chu think about this @mattkrick?

File: src/server/graphql/models/User/userMutation.js [hunk 2]
```diff
@@ -168,26 +168,37 @@ export default {
...
       // RESOLUTION
       // propagate denormalized changes to TeamMember
-      const dbProfile = await r.table('TeamMember')
+      const dbProfileChanges = await r.table('TeamMember')
         .getAll(id, {index: 'userId'})
         .update(validUpdatedUser)
         .do(() => {
           return r.table('User')
             .get(id)
             .update(validUpdatedUser, {returnChanges: 'always'});
```
Comment (mattkrick) [inline src/server/graphql/models/User/userMutation.js]: we should probably update the `updatedAt` field, which would guarantee that we return a change, which means we can return `('changes')(0)('new_val')`. but we can do that after the merge

File: src/universal/modules/notifications/components/TrialExpired/TrialExpired.js [hunk 3]
```diff
@@ -8,10 +8,14 @@ import Row from 'universal/components/Row/Row';
 import FontAwesome from 'react-fontawesome';
 import appTheme from 'universal/styles/theme/appTheme';
 import ui from 'universal/styles/ui';
+import {segmentEventTrack} from 'universal/redux/segmentActions';
 
-const TrialExpiresSoon = (props) => {
-  const {orgId, router, styles} = props;
+const TrialExpires = (props) => {
```
Comment (mattkrick) [inline src/universal/modules/notifications/components/TrialExpired/TrialExpired.js]: rename to `TrialExpired`?

File: src/universal/modules/userDashboard/containers/CreditCardModal/CreditCardModalContainer.js [hunk 4]
```diff
@@ -89,6 +94,7 @@ class CreditCardModalContainer extends Component {
       };
       const {error: cashayError} = await cashay.mutate('addBilling', {variables});
       if (cashayError) throw new SubmissionError(cashayError);
+      dispatch(segmentEventTrack('addStripeBilling Success'));
```
Comment (mattkrick) [inline src/universal/modules/userDashboard/containers/CreditCardModal/CreditCardModalContainer.js]: where'd the cashay error to segment go? i don't see any changes to `addBilling.js` on the server. Should we add that & this success to that mutation?

Comment (jordanh) [inline src/universal/modules/userDashboard/containers/CreditCardModal/CreditCardModalContainer.js]: Ah, good ideas. I will do that.

File: src/universal/modules/notifications/components/TrialExpired/TrialExpired.js (hunk 3 above)
Comment (jordanh) [inline src/universal/modules/notifications/components/TrialExpired/TrialExpired.js]: Good catch! I tried to make it better, and made it worse ;)

File: src/server/graphql/models/Organization/addBilling/addBilling.js [hunk 5]
```diff
@@ -107,6 +126,16 @@ export default {
             })
             .delete();
         });
+      if (segmentIo) {
```
Comment (mattkrick) [inline src/server/graphql/models/Organization/addBilling/addBilling.js]: dupe of line 56?

think it's worthwhile to mock segmentIo if `SEGMENT_WRITE_KEY` doesn't exist? then we wouldn't have to wrap every call in a conditional

**Comment by jordanh** [comment]
Okee dokey. All the changes are in. Merging this sucker now.

**Pull request merged**